src/coreclr/vm/loongarch64/asmhelpers.S

```diff
@@ -197,28 +201,35 @@ WRITE_BARRIER_ENTRY JIT_WriteBarrier
 
     st.d  $t7, $t6, 0
 
+    // load the address wbs_card_table to the $t3
+    // and will cache the wbs_card_table by $t3 within the JIT_WriteBarrier.
+    // So please DONT'T overwrite the $t3 within the JIT_WriteBarrier !!!
+.Ltmp1:
+    pcaddi $t3, 0
+    .reloc .Ltmp1, R_LARCH_PCREL20_S2, wbs_card_table
+
 #ifdef WRITE_BARRIER_CHECK
     // Update GC Shadow Heap
 
-    // Do not perform the work if g_GCShadow is 0
-    la.local $t1, wbs_GCShadow
-    ld.d $t1, $t1, 0
+    // the wbs_GCShadow = $t3 + 16
+    ld.d $t1, $t3, 16
 
+    // Do not perform the work if g_GCShadow is 0
     beq  $t1, $zero, 22f //LOCAL_LABEL(ShadowUpdateDisabled)
 
     // Compute address of shadow heap location:
     //   pShadow = g_GCShadow + ($t6 - g_lowest_address)
-    la.local  $t3, wbs_lowest_address
-    ld.d $t3, $t3, 0
+    // the wbs_lowest_address = $t3 + 48
+    ld.d $t0, $t3, 48
 
-    sub.d  $t3, $t6, $t3
-    add.d  $t0, $t3, $t1
+    sub.d  $t0, $t6, $t0
+    add.d  $t0, $t0, $t1
 
     // if (pShadow >= g_GCShadowEnd) goto end
-    la.local $t3, g_GCShadowEnd
-    ld.d  $t3, $t3, 0
+    la.local $t1, g_GCShadowEnd
```

Comment:
Does this has the same problem as what was fixed in https://github.com/dotnet/runtime/pull/90036? Should `g_GCShadowEnd` be cloned in the `wbs_card_table` too?

Reply:
Thanks.
For our tests, this is ok.
I will check it again.

Reply:
Thanks very much.
I add this.